### PR TITLE
Add Nexus OutgoingService CRUD

### PR DESCRIPTION
**Pull request by MichaelSnowden**
**What changed?**
Add OperatorService APIs for managing Nexus services to which we will send traffic.

**Why?**

**Breaking changes**

### CODE REVIEW

File: temporal/api/operatorservice/v1/service.proto [hunk 1]
```diff
@@ -101,4 +97,31 @@ service OperatorService {
...
+    // need to increment the version yourself. The server will increment the version for you after each update.
+    rpc CreateNexusOutgoingService(CreateNexusOutgoingServiceRequest) returns (CreateNexusOutgoingServiceResponse) {
+    }
+
+    // Update an outgoing Nexus service by namespace and service name. The version in the request should match the
+    // current version of the service. This will fail with a status of FAILED_PRECONDITION if the version does not match.
+    // Returns the updated service with the updated version, which can be used for subsequent updates. You don't need
+    // to increment the version yourself. The server will increment the version for you.
+    rpc UpdateNexusOutgoingService(UpdateNexusOutgoingServiceRequest) returns (UpdateNexusOutgoingServiceResponse) {
+    }
```
[the remark this answers is not in the document]
Comment (bergundy) [reply temporal/api/operatorservice/v1/service.proto:118]: We'll also separate these for incoming in a followup PR.

File: api-linter.yaml
[the remark this answers is not in the document]
Comment (MichaelSnowden) [reply api-linter.yaml:1]: Yes, this is meant to be a stacked PR. Should've put that in the description.

**Comment by MichaelSnowden** [comment]
> The protos LGTM, please separate the linter change into a different PR.

Rebased with those on master now.

File: temporal/api/operatorservice/v1/service.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (cretz) [reply temporal/api/operatorservice/v1/service.proto:118]: If we know we're going to do it, might as well do it together IMO.

The separate PRs makes it hard to review as a whole. I was lucky enough have looked at the other Nexus APIs for consistency or I would have missed this inconsistency entirely (even though I'm sure y'all knew about it, it was unclear to me).

File: temporal/api/operatorservice/v1/service.proto [hunk 2]
```diff
@@ -97,4 +97,31 @@ service OperatorService {
...
+
+    // Delete an outgoing Nexus service by namespace and service name.
+    rpc DeleteNexusOutgoingService(DeleteNexusOutgoingServiceRequest) returns (DeleteNexusOutgoingServiceResponse) {
+    }
+
+    // List all Nexus outgoing services for a namespace. Use next_page_token in the response for pagination.
+    rpc ListNexusOutgoingServices(ListNexusOutgoingServicesRequest) returns (ListNexusOutgoingServicesResponse) {
```
Comment (bergundy) [inline temporal/api/operatorservice/v1/service.proto:125]: Might want to mention the pagination behavior here.
E.g. what happens if the list is mutated while paging.

**Review by bergundy — APPROVED** [review]

File: temporal/api/operatorservice/v1/service.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (bergundy) [reply temporal/api/operatorservice/v1/service.proto:118]: I'll submit a PR to follow this one once it's merged.

File: temporal/api/operatorservice/v1/service.proto [hunk 3]
```diff
@@ -97,4 +97,34 @@ service OperatorService {
...
+    }
+
+    // Delete an outgoing Nexus service by namespace and service name.
+    rpc DeleteNexusOutgoingService(DeleteNexusOutgoingServiceRequest) returns (DeleteNexusOutgoingServiceResponse) {
+    }
+
+    // List all Nexus outgoing services for a namespace. Populate page_token in the request from the next_page_token
+    // field of a previous response to paginate through all Nexus outgoing services. An empty next_page_token indicates
+    // that there are no more services after the last one in the current response. If the list is modified during
+    // pagination, the result is undefined.
```
Comment (bergundy) [inline temporal/api/operatorservice/v1/service.proto]: Maybe there's a better way to phrase this but might be worth calling out:

```suggestion
    // List all Nexus outgoing services for a namespace in ascending name order. Populate page_token in the request from the next_page_token
    // field of a previous response to paginate through all Nexus outgoing services. An empty next_page_token indicates
    // that there are no more services after the last one in the current response. If the list is modified during
    // pagination, the next page will start from a service with a name that has a higher lexicographical value than the last service in the previous page.
```

Comment (bergundy) [reply temporal/api/operatorservice/v1/service.proto]: Nice, you phrased it better than I did.

File: temporal/api/operatorservice/v1/service.proto (hunk 2 above)
Comment (MichaelSnowden) [reply temporal/api/operatorservice/v1/service.proto:128]: Done

**Review by cretz — APPROVED** [review]

**Pull request merged**